components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityUtil.java

```diff
@@ -2254,4 +2259,113 @@ public static String getAgentIdentityUserstoreName() {
         }
         return userStoreName;
     }
+
+    /**
+     * Check whether the JWT exceeds the allowed depth.
+     *
+     * @param jwt JWT string to check.
+     * @throws  ParseException If the JWT exceeds the allowed depth or if an error occurs during parsing.
+     */
+    public static void validateJWTDepth(String jwt) throws ParseException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Checking JWT depth validation");
+        }
+
+        if (StringUtils.isBlank(jwt)) {
+            if (log.isDebugEnabled()) {
+                log.debug("JWT is blank, skipping depth validation");
+            }
+            throw new ParseException("Error validating JWT depth. JWT is blank.", 0);
+        }
+
+        // Extract and decode JWT payload.
+        String[] parts = jwt.split("\\.");
```

Comment:
If we consider jwt here, we issue jwts separating with `.`. So considering splitted array size to be one will be a problem. 

Check choreo login access token for example

Reply:
size being one will be rejected with the below if condition